### PR TITLE
feat(telemetry): Adding Scarf based telemetry to Superset

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

This PR does a handful of things:
* Adds the Scarf pixel to Superset (as approved by SIP) via a new `TelemetryPixel` component
* Adds the `ENABLE_TELEMETRY` feature flag 
* Adds basic tests for the above
* Bumps react types to support the `no-referrer-when-downgrade` 
* Bumps the npm version... not that I intended to, but it keeps coming along for the ride
* Fixes the shim for frontend tests, which had a weird incompatibility with certain node 18.x versions, throwing a `TypeError: Cannot redefine property: window` error

TODO:
* Get the pixel to load via Scarf gateway so that url params (version/sha/build) are reliably captured
* Feature flag documentation
* Turn off _more_ Scarf stuff with this flag?

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

RTL tests for the new component with FF on and off should pass. Run it, and have any PMC member check the Scarf data for proper capture. 

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/25639
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/features/home/RightMenu.tsx [hunk 1]
```diff
@@ -474,7 +475,7 @@ const RightMenu = ({
             <Menu.ItemGroup key="user-section" title={t('User')}>
               {navbarRight.user_profile_url && (
                 <Menu.Item key="profile">
-                  <Link to={navbarRight.user_profile_url}>{t('Profile')}</Link>
+                  <a href={navbarRight.user_profile_url}>{t('Profile')}</a>
```
Comment (john-bodley) [inline superset-frontend/src/features/home/RightMenu.tsx:478]: Should we preserve the native React component?

Comment (rusackas) [reply superset-frontend/src/features/home/RightMenu.tsx:478]: Not sure how this happened, but nice catch, you Javascript enthusiast!

**Comment by michael-s-molina** [comment]
The [John ❤️ JS commit](https://github.com/apache/superset/pull/26011/commits/925ef36d8f91d3302405bda80094f0bb4d701b51) cracked me up! 🤣

File: superset-frontend/src/components/TelemetryPixel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * @param {string} props.sha - The SHA of Superset that's currently in use.
+ * @param {string} props.build - The build of Superset that's currently in use.
+ * @returns {JSX.Element | null} The rendered TelemetryPixel component.
+ */
+const TelemetryPixel = ({
+  version = 'unknownVersion',
+  sha = 'unknownSHA',
+  build = 'unknownBuild',
+}: TelemetryPixelProps): JSX.Element | null => {
+  const pixelId = '0d3461e1-abb1-4691-a0aa-5ed50de66af0';
```
Comment (villebro) [inline superset-frontend/src/components/TelemetryPixel/index.tsx:45]: nit: maybe we could consider moving this outside the component and renaming it `PIXEL_ID` as it's really a constant.

**Review by villebro — APPROVED** [review]
LGTM with a minor nit, looking forward to this 👍

File: superset-frontend/packages/superset-ui-core/src/utils/featureFlags.ts [hunk 3]
```diff
@@ -44,6 +44,7 @@ export enum FeatureFlag {
   ENABLE_ADVANCED_DATA_TYPES = 'ENABLE_ADVANCED_DATA_TYPES',
   ENABLE_EXPLORE_DRAG_AND_DROP = 'ENABLE_EXPLORE_DRAG_AND_DROP',
   ENABLE_JAVASCRIPT_CONTROLS = 'ENABLE_JAVASCRIPT_CONTROLS',
+  ENABLE_TELEMETRY = 'ENABLE_TELEMETRY',
```
Comment (michael-s-molina) [inline superset-frontend/packages/superset-ui-core/src/utils/featureFlags.ts:47]: Can we remove this now that we depend on an environment variable?

File: superset-frontend/src/components/TelemetryPixel/TelemetryPixel.test.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { render } from 'spec/helpers/testing-library';
+import TelemetryPixel from '.';
+
+describe('environmental variables', () => {
```
Comment (michael-s-molina) [inline superset-frontend/src/components/TelemetryPixel/TelemetryPixel.test.tsx:23]: Can you [avoid the describe block](https://github.com/apache/superset/wiki/Testing-Guidelines-and-Best-Practices#avoid-nesting-when-youre-testing) here?

File: superset-frontend/packages/superset-ui-core/src/utils/featureFlags.ts (hunk 3 above)
Comment (rusackas) [reply superset-frontend/packages/superset-ui-core/src/utils/featureFlags.ts]: LOL you're on top of it! Done, and more cleanup to come.

File: superset-frontend/src/components/TelemetryPixel/TelemetryPixel.test.tsx (hunk 4 above)
Comment (rusackas) [reply superset-frontend/src/components/TelemetryPixel/TelemetryPixel.test.tsx]: Also done. Docs forthcoming, then I think we'll be good.

File: superset-frontend/src/components/TelemetryPixel/index.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  sha = 'unknownSHA',
+  build = 'unknownBuild',
+}: TelemetryPixelProps): React.ReactElement | null => {
+  const pixelPath = `https://apachesuperset.gateway.scarf.sh/pixel/${PIXEL_ID}/${version}/${sha}/${build}`;
+
+  console.warn(
+    'scarf',
+    process.env.SCARF_ANALYTICS,
+    typeof process.env.SCARF_ANALYTICS,
+  );
```
Comment (michael-s-molina) [inline superset-frontend/src/components/TelemetryPixel/index.tsx:53]:
```suggestion
```

File: superset-frontend/src/components/TelemetryPixel/index.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
```
Comment (michael-s-molina) [inline superset-frontend/src/components/TelemetryPixel/index.tsx:54]:
```suggestion
```

**Review by michael-s-molina — APPROVED** [review]
LGTM

File: superset-frontend/src/components/TelemetryPixel/index.tsx (hunk 6 above)
Comment (rusackas) [reply superset-frontend/src/components/TelemetryPixel/index.tsx]: Hah, thanks!

File: superset-frontend/src/components/TelemetryPixel/index.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+const PIXEL_ID = '0d3461e1-abb1-4691-a0aa-5ed50de66af0';
+
+const TelemetryPixel = ({
+  version = 'unknownVersion',
+  sha = 'unknownSHA',
+  build = 'unknownBuild',
+}: TelemetryPixelProps): React.ReactElement | null => {
+  const pixelPath = `https://apachesuperset.gateway.scarf.sh/pixel/${PIXEL_ID}/${version}/${sha}/${build}`;
+  return process.env.SCARF_ANALYTICS === 'false' ||
+    process.env.SCARF_ANALYTICS === 'false' ? null : (
```
Comment (michael-s-molina) [inline superset-frontend/src/components/TelemetryPixel/index.tsx:49]:
```
process.env.SCARF_ANALYTICS === 'false' ||
    process.env.SCARF_ANALYTICS === 'false'
```
Just noticed this is duplicated.

Comment (rusackas) [reply superset-frontend/src/components/TelemetryPixel/index.tsx:49]: Yeesh... I thought I nabbed that. Will open a PR to clean up... Especially if this does not have the intended effect now. Surprised some linting rule doesn't spot this kind of thing.

**Pull request merged**